File: conformance/bundles/05-ambiguous-match/CukesSteps.java
package dev.varar.conformance.bundle05;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;

/**
 * Java sibling of {@code cukes.steps.ts} / {@code cukes.steps.py} (bundle {@code
 * 05-ambiguous-match}). Both expressions match "I have 5 cukes" -&gt; ambiguous-match
 * diagnostic (a later, plan-stage concern; this stage only needs both registered).
 */
public final class CukesSteps implements StepDefinitions<CukesSteps.Ctx> {

    record Ctx() implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.defineState(Ctx::new);

        s.stimulus("I have {int} cukes", (Ctx ctx, Integer n) -> ctx);
        s.stimulus("I have 5 cukes", (Ctx ctx) -> ctx);
    }
}

File: conformance/bundles/05-ambiguous-match/cukes.steps.kt
@file:JvmName("CukesSteps")

// Kotlin sibling of cukes.steps.ts / cukes.steps.py / CukesSteps.java (bundle
// 05-ambiguous-match): both expressions match "I have 5 cukes" -> ambiguous-
// match diagnostic at the plan stage; this stage only needs both registered.
package dev.varar.kotlin.conformance.bundle05

import dev.varar.kotlin.stimulus
import dev.varar.kotlin.steps

class Ctx

val steps = steps(::Ctx) {
    stimulus("I have {int} cukes") { n: Int -> this }
    stimulus("I have 5 cukes") { this }
}
